src/lib/autodiscovery/autodiscovery.py

```diff
@@ -18,6 +20,66 @@
 discovered_resource_type = os.environ.get("AUTODISCOVERY_RESOURCE_TYPE", "gce_instance")
 
 
+async def send_metadata(missing_metrics_list: List[Metric], gcp_session: ClientSession, token: str):
+    to_send = []
+
+    for metric in missing_metrics_list:
```

Comment:
I would wrap entire for loop into the function and you can make it tore readable like 
metrics_to_send = extract_missing_metrics(missing_metrics_list) //naming up to you